### DISCUSSION TITLE
a question about link prediction

### DISCUSSION THREAD

**Discussion by HanChen-HUST** [pyg-team/pytorch_geometric#6028] [Q&A]
![image](https://user-images.githubusercontent.com/45957029/203074525-75d4ae51-7428-4714-abea-89bd3f198371.png)
![image](https://user-images.githubusercontent.com/45957029/203074664-c26a1ba9-b5ad-439e-b43f-6a3b66275ab2.png)

it can work on Plantoid dataset,but could not work on Amazon Dataset?what is the reason?Thanks@rusty1s

**Comment by EdisonLeeeee** [comment]
Which PyG version are you using? There was a bug in `RandomLinkSplit` but recently solved in master.

**Comment by HanChen-HUST** [comment]
when I set the device to cpu,it could work,but the running speed is too slow,how can I solve it？

**Reply by EdisonLeeeee** [reply]
Seems related to https://github.com/pyg-team/pytorch_geometric/issues/5721#issuecomment-1284225736
